src/me/remie/vulcan/leaguetasks/task/tasks/fishing/FishAnchovies.java

```diff
@@ -0,0 +1,183 @@
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areWeInCatherbyRegion() {
+        final int playerCurrentRegion = ctx.players.getLocal().getLocation().getRegionID();
+        if (playerCurrentRegion != LeagueScriptConstants.CATHERBY_REGION_ID && playerCurrentRegion != FISHING_SPOT_REGION) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean openUpBank() {
+        if (!ctx.bank.bankOpen()) {
+            script.setScriptStatus("Opening up bank");
+            final SimpleNpc banker = ctx.npcs.populate().filter(LeagueScriptConstants.BANKER_NPC_ID, 1613).nextNearest();
+            if (banker != null) {
+                banker.menuAction("Bank");
+                return ctx.onCondition(ctx.bank::bankOpen, 250, 10);
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private boolean depositInventory() {
+        if (!openUpBank()) {
+            return false;
+        }
+
+        if (!ctx.bank.bankOpen()) {
+            return false;
+        }
+
+        script.setScriptStatus("Depositing inventory");
+        ctx.bank.depositAllExcept(LeagueScriptConstants.COINS_ID, SMALL_FISHING_NET.getItemId());
+        ctx.sleep(1000);
+        script.setScriptStatus("Closing bank");
+        ctx.bank.closeBank();
+        ctx.sleep(1000);
+        return true;
+    }
+
+    private boolean getRequiredItem() {
+        if (!ctx.inventory.populate().omit(LeagueScriptConstants.COINS_ID).omit(SMALL_FISHING_NET.getItemId()).isEmpty()) {
+            return depositInventory();
+        }
+        script.setScriptStatus("Acquiring " + SMALL_FISHING_NET.getItemName());
+        if (fishingNetShopHelper.buyItem(SMALL_FISHING_NET, ShopType.GENERAL_STORE)) {
+            ctx.sleep(1000);
+        }
+        return true;
+    }
+
+    private boolean navigateToFishingSpot() {
+        script.setScriptStatus("Navigating to fishing spot");
+        return ctx.onCondition(() -> ctx.pathing.walkPath(catherbyBankToFishingSpotPath), 400, 20);
+    }
+
+    private boolean doWeHaveRequiredItem() {
+        return !ctx.inventory.populate().filter(SMALL_FISHING_NET.getItemId()).isEmpty();
+    }
+
+    private boolean fishAnchovies() {
+        if (ctx.players.getLocal().getAnimation() != -1 || ctx.inventory.inventoryFull()) {
+            return true;
+        }
+
+        script.setScriptStatus("Fishing");
+        final SimpleNpc fishSpot = ctx.npcs.populate().filter(FISHING_SPOT_NPC_ID).nextNearest();
+        if (fishSpot != null) {
+            fishSpot.menuAction("Small Net");
+            ctx.sleep(1000);
+            return true;
+        }
+        return false;
+    }
+
+    private boolean depositItemsInCatherbyBank() {
+        script.setScriptStatus("Navigate to bank");
+        if (ctx.onCondition(() -> ctx.pathing.walkPath(catherbyBankToFishingSpotPath, true), 600, 20)) {
```

Comment:
WalkPath only steps to one tile at a time, it won't walk the whole path, can just call the method once instead of holding the thread with the onCondition.